app/models/pglogical_subscription.rb

```diff
@@ -28,6 +28,19 @@ def self.lookup_by_id(to_find)
     nil
   end
 
+  def self.with_connection_error_handling
+    retry_attempted = false
```

Comment:
I think this is incorrect.  If you `retry` it will also re-execute this line setting retry_attempted back to false. I think you can eliminate this line altogether, because by default the value will be nil which is falsey.

Reply:
Yeah, that sounds right... or use a `||=` operator, which shouldn't change anything on the second pass.  I think that was what I wanted to do in my head, but it didn't come out in the code.

Anyway, will change.  Thanks!  👍

Reply:
So, just to clarify:

> I think you can eliminate this line altogether, because by default the value will be nil which is falsey.

This isn't true, as it does end up raising an error:

```console
$ ruby reraise.rb

Traceback (most recent call last):
        2: from reraise.rb:15:in `<main>'
        1: from reraise.rb:6:in `bar'
reraise.rb:2:in `foo': NoMethodError (NoMethodError)
        2: from reraise.rb:15:in `<main>'
        1: from reraise.rb:5:in `bar'
reraise.rb:8:in `rescue in bar': undefined local variable or method `retry_attempted' for main:Object (NameError)
```

But the `||=` does work as expected.